### PR TITLE
generate kube

### PR DESCRIPTION
**Pull request by baude**
add the ability to generate kubernetes pod and service yaml representations
of libpod containers and pods.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podman/generate_kube.go [hunk 1]
```diff
@@ -59,33 +60,43 @@ func generateKubeYAMLCmd(c *cli.Context) error {
...
-	developmentComment := []byte("# Generation of Kubenetes YAML is still under development!\n")
-	logrus.Warn("This function is still under heavy development.")
-	// Marshall the results
-	b, err := yaml.Marshal(podYAML)
-	if err != nil {
-		return err
-	}
-	output = append(output, developmentComment...)
-	output = append(output, b...)
+	header := `# Generation of Kubenetes YAML is still under development!
```
Comment (TomSweeneyRedHat) [inline cmd/podman/generate_kube.go]: are you missing and ending `  ?

File: docs/podman-generate-kube.1.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
+% PODMAN(1) Podman Man Pages
```
Comment (TomSweeneyRedHat) [inline docs/podman-generate-kube.1.md]: Thought this line was supposed to have generate in it....

File: docs/podman-generate.1.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+% podman-generate(1)
```
Comment (TomSweeneyRedHat) [inline docs/podman-generate.1.md]: Like this for the previous page

File: cmd/podman/generate_kube.go (hunk 1 above)
Comment (giuseppe) [inline cmd/podman/generate_kube.go]: s|Kubenetes|Kubernetes|

File: cmd/podman/generate_kube.go [hunk 4]
```diff
@@ -18,16 +19,15 @@ var (
...
 	containerKubeCommand     = cli.Command{
-		Name:                   "generate",
-		Usage:                  "Generate Kubernetes pod YAML for a container",
+		Name:                   "kube",
+		Usage:                  "Generate Kubernetes pod YAML for a container or pod",
 		Description:            containerKubeDescription,
 		Flags:                  sortFlags(containerKubeFlags),
 		Action:                 generateKubeYAMLCmd,
-		ArgsUsage:              "CONTAINER-NAME",
+		ArgsUsage:              "CONTAINER|POD-NAME",
```
Comment (giuseppe) [inline cmd/podman/generate_kube.go]: I'd make it explicit whether it is a container or a pod (i.e. add a `--pod` flag if we decide to use containers by default, or `--container` if the other way around).

Otherwise how to pick the pod if we have a pod and a container with the same name?

File: cmd/podman/generate_kube.go [hunk 5]
```diff
@@ -36,20 +36,21 @@ var (
...
+		container      *libpod.Container
+		err            error
+		output         []byte
+		pod            *libpod.Pod
+		mashalledBytes []byte
+		servicePorts   []v1.ServicePort
 	)
 
 	if rootless.IsRootless() {
 		return errors.Wrapf(libpod.ErrNotImplemented, "rootless users")
```
Comment (giuseppe) [inline cmd/podman/generate_kube.go]: not part of this PR, but I think it is safe to drop this check.  Rootless containers should just work.

File: cmd/podman/generate_kube.go (hunk 4 above)
Comment (mheon) [reply cmd/podman/generate_kube.go]: That's actually not allowed by design - containers and pods share the same pool of names, so you can't have a container and pod with the same name.

**Comment by mheon** [comment]
LGTM

**Comment by mheon** [comment]
@rhatdan PTAL
